salt/orch/force_removal.sls

```diff
@@ -0,0 +1,72 @@
+# This is a best effort orchestration. Any operation run on the minions by this orchestration
+# could fail, as we don't know the current state of the cluster at this point. The idea is to
+# try to clean as much as possible where possible, but in the end the only thing we can really
+# ensure is that the salt-key of the target is rejected.
+
+# must provide the node (id) to be removed in the 'target' pillar
+{%- set target = salt['pillar.get']('target') %}
+{%- set target_nodename = salt.saltutil.runner('mine.get', tgt=target, fun='nodename')[target] %}
+# services that will be called when trying to unregister as clean as possible
+{%- set services_to_stop = { "kubelet": ["stop", "remove-pre-reboot"], "etcd": ["remove-pre-stop-services", "remove-pre-reboot"] } %}
+# nodes that are responding
+{%- set up_nodes = salt.saltutil.runner('manage.up', tgt='P@roles:kube-* and not ' + target, expr_form='compound') %}
```

Comment:
It would be better to get the list of nodes that are down (it should be short) instead of the list of nodes that are up (it can be very long). Not necessary to get this list anyway...

Reply:
I will have a look at how I can improve this.